crates/storage_models/src/payment_intent.rs

```diff
@@ -122,6 +125,7 @@ pub struct PaymentIntentUpdateInternal {
     pub billing_address_id: Option<String>,
     pub shipping_address_id: Option<String>,
     pub modified_at: Option<PrimitiveDateTime>,
+    pub attempt_id: String,
```

Comment:
Make this an `Option<String>` instead. When you use any of the enums variants other than `PaymentAttemptUpdate`, you'll end up setting the `attempt_id` to an empty `String` because of the `Default::default()` being called.

Reply:
Yes. 

Apologies this change was pushed accidentally. I made the same code changes in `for-retry` branch and already used `Option` there.